src/ts/ui/App.ts

```diff
@@ -0,0 +1,70 @@
+import ProductManagementDomain from '../domain/ProductManagement';
+import CoinManagementDomain from '../domain/CoinManagement';
+import { $, $$ } from '../utils/dom';
+import CoinManagementUI from './CoinManagementUI';
+import ProductManagementUI from './ProductManagementUI';
+import ProductPurchaseUI from './ProductPurchase';
+
+export default class App {
+  private productDomain;
+  private coinDomain;
+  private productManagementUI;
+  private coinManagementUI;
+  private productPurchaseUI;
+
+  constructor() {
+    this.productDomain = new ProductManagementDomain();
+    this.coinDomain = new CoinManagementDomain();
+
+    this.productManagementUI = new ProductManagementUI(this.productDomain);
```

Comment:
```suggestion
  constructor() {
    private readonly productDomain = new ProductManagementDomain(),
    private readonly coinDomain = new CoinManagementDomain(),
    private readonly productManagementUI = new ProductManagementUI(productDomain)
    // ...
```

생성자에서 초기화해보면 어떨까요?
클래스에 대한 인터페이스에 대해 타입힌팅을 받을 수 있어요.

컴파일 결과가 궁금하시다면 간단한 예시의 [플레이그라운드](https://www.typescriptlang.org/play?#code/MYGwhgzhAEBSCmATAltA3tAsAKGn6wA9gHYQAuATgK7BmEUAUO+L0ADhcgG5hnzQV4YRCRABPaMTABbeAC5o5TsQDm0ALzQA5AGUA1mIDuYEHvgUtzfAEp0AXxxW8dHZWSqG1uUvdq0T1kEyKgpiaDIAC2QIADopWQBuAIdsFJweCmgAKyRUTWJ4QzhczyTsHCJSQhB4GJBCFQYclBiXNw9rawToAHoe-SMTMwogA)를 참고해주세요

Reply:
확실히 생성자에서 초기화하니 자동으로 타입이 추론되네요..! any가 뜨지 않는다는게 이렇게 편안할 줄이야..

근데 플레이그라운드와 위에 예시로 올려주신 코드가 초기화 선언 위치가 다른 것 같아서요. 직접 해보니 에러가 나지 않는 쪽은 플레이그라운드의 방식대로 () 안에서 선언해주는 방식이었는데, 혹시 {} 안에서 선언해주면서 에러가 나지 않는 방법이 또 있는걸까요?

Reply:
옮기면서 실수를한거같아요! ()안이 맞습니다 😅 

생성자에서 하는수밖에 없을거에요!...
중괄호에서 하면 클래스 필드를 선언하고 타입까지 명시해줘야해요 😢 

오류 관련해서 코멘트를 주셔서 플레이그라운드를 이용하다가 끄적거린건데 혹시몰라 남겨드려요 👀 
```typescript
interface PersonProps { 
    name?: string 
    age?: number;    
}

class Person { 
    constructor(
        readonly state : PersonProps = { name: 'Asa', age: 99999 }
    ) {}

    toString() {
        return `나이는 ${this.state.age}이고 이름은 ${this.state.name}입니다.`;
    }
}

const asa = new Person();

console.log(asa.toString()); // 나이는 99999이고 이름은 Asa입니다. 
```

Reply:
예시 코드까지..! 감사합니다 :)

알려주신 방식대로 수정 완료하였습니다! https://github.com/woowacourse/javascript-vendingmachine/pull/12/commits/f788061f651245e1a48086789151b6aefca493cf

다만 코드를 읽으면서 readonly에 대해 한 가지 궁금한 점이 생겼습니다. readonly는 읽기 전용으로 만들어줄 때 붙여주는 것으로 알고 있는데요, 수정할 일이 없는 요소에는 전부 readonly를 붙여주는 것이 좋을까요? 아직 readonly를 제대로 사용한 적이 없다보니, 언제 써야할지 감이 잘 안 잡히는 것 같아요 💦

Reply:
readonly는 인스턴스를 다시 할당하고 싶지 않을 때 사용하면 되요.

Angular, NestJS에서 service를 di로 주입할 때 private readonly를 사용하더라구요.
NestJS를 조금 다뤄보다보니 생성자에서 readonly를 익숙하게 사용하고 있어요.
생성자에서 초기화 한 인스턴스를 굳이 재할당을 할 가능성이 높지않아 유의미한지는 저도 확신이 안서네요 😅